modules/async/async/shared/src/main/scala/async.scala

```diff
@@ -33,6 +34,16 @@ object async {
     def async[A](fa: Proc[A]): FS[A]
   }
 
+  class Future2AsyncM[F[_]](implicit AC: AsyncContext[F], E: ExecutionContext)
+    extends FSHandler[Future, F] {
+    override def apply[A](future: Future[A]): F[A] =
+      AC.runAsync { cb =>
+        E.execute(new Runnable {
+          def run(): Unit = future.onComplete(_.toEither)
```

Comment:
Not sure if `Try#toEither` was backported to 2.11. If it works great otherwise pattern matching it is.

Reply:
Sadly it is not backported to 2.11.x.